File: backend/src/main/java/Pracownia/Projekt/Spring/Entities/BookCopy.java
package Pracownia.Projekt.Spring.Entities;

import Pracownia.Projekt.Spring.Constants.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class BookCopy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    @CreationTimestamp
    private Date createdAt;

    @Column
    private Status status;

    @ManyToOne
    private Book book;

}

File: frontend/src/models/BookCopy.ts
import { Status } from "../constants/enums"

export interface BookCopy {
    id: number,
    createdAt: Date,
    status: Status,
    title: string,
    author: AuthorShort

}

export interface AuthorShort {
    id: number,
    firstName: string,
    lastName: string
}